### PR TITLE
Add documentation to vis_augmenter

### LINKED ISSUE #3383: [Feature Anywhere] [Meta] Add documentation for Feature Anywhere
Coming from #2880

We will want to have clear documentation in READMEs to show the following:

- [x] purpose of the `vis_augmenter` plugin and how it works
- [x] examples of each added data model (a `VisLayer`, a `PluginResource`, an `augment-vis` saved object, a `vis_layer` expressions function)
- [x] detailed steps on how to onboard a plugin. This should live in the `vis_augmenter` README, as well as a standalone blog. It may reference plugin documentation with examples
- [x] examples, usage, & purpose of the new `vis_augmenter/stats` API
- [x] limitations / gotchas of the feature (e.g., default limit of 10 resources per vis)
- [x] the loose coupling, such as freedom to make visualizations ineligible, add plugin resources that are unrelated to the vis, etc.

### PR DESCRIPTION
**Pull request by ohltyler**
### Description

Add documentation to vis_augmenter plugin.

### Issues Resolved

closes #3383 

### Check List

- [ ] All tests pass
  - [ ] `yarn test:jest`
  - [ ] `yarn test:jest_integration`
  - [ ] `yarn test:ftr`
- [ ] New functionality includes testing.
- [ ] New functionality has been documented.
- [ ] Update [CHANGELOG.md](./../CHANGELOG.md)
- [ ] Commits are signed per the DCO using --signoff

### CODE REVIEW

**Comment by ohltyler** [comment]
Leaving as draft for now. Need to check all boxes as specified in #3383

File: src/plugins/vis_augmenter/README.md [hunk 1]
```diff
@@ -1 +1,71 @@
...
+
+### Overview
+
+The feature introduced in [#4361](https://github.com/opensearch-project/OpenSearch-Dashboards/pull/4361) allows for plugins to be able to display additional data on a visualization. The framework for how that is done, including all of the interfaces, type definitions, helper functions, and services used, are maintained in this plugin. It also registers new saved object types and expression types that plugins use to standardize how data is formatted and visualized on the charts.
+
+### Eligibility
+
+Currently, this augmenting framework is only eligible for line visualizations, with a single date histogram x axis. There is future plans to expand this to other types of visualizations, as well as other ways that visualizations may be augmented, such as overlaying forecasting data.
+
+### Framework
```
Comment (ashwin-pc) [inline src/plugins/vis_augmenter/README.md:11]: nit: Is there a visual bock diagram that you can attach that outlines this framework? The description is nice but visuals will help understand the mental model better :)

**Review by ashwin-pc — APPROVED** [review]
@ohltyler This is really nice! great job documenting this stuff. Save for my nit about adding some images here, the rest looks good :) If we expand this in future, lets also consider creating an example plugin that utilizes this since augmenting visualizations seems to be something that a lot of people are interested in. But i'm getting ahead of myself.

File: CHANGELOG.md [hunk 2]
```diff
@@ -61,6 +61,7 @@ Inspired from [Keep a Changelog](https://keepachangelog.com/en/1.0.0/)
 - [Doc] Add COMMUNICATIONS.md with info about Slack, forum, office hours ([#3837](https://github.com/opensearch-project/OpenSearch-Dashboards/pull/3837))
 - [Saved Object Service] Adds design doc for new Saved Object Service Interface for Custom Repository [#3954](https://github.com/opensearch-project/OpenSearch-Dashboards/pull/3954)
 - Add plugin development section in DEVELOPER_GUIDE.md ([#778](https://github.com/opensearch-project/OpenSearch-Dashboards/pull/3989))
+- Add documentation to vis_augmenter ([#4527](https://github.com/opensearch-project/OpenSearch-Dashboards/pull/4527))
```
Comment (joshuarrrr) [inline CHANGELOG.md:64]:
```suggestion
- [Vis Augmenter] Add documentation to `vis_augmenter` plugin ([#4527](https://github.com/opensearch-project/OpenSearch-Dashboards/pull/4527))
```

**Review by joshuarrrr — APPROVED** [review]
Nice - this is great to have for other builders to think about how to leverage the new capability. If we see increased interest, the next step would be to add developer examples with a minimal plugin integration.

**Comment by joshuarrrr** [comment]
@ohltyler Because this is dev documentation, merging to `main` is likely more important than including it in the distribution of 2.9.0, which is already code frozen. OK relabeling for 2.10.0 backport?

**Comment by ohltyler** [comment]
> @ohltyler Because this is dev documentation, merging to `main` is likely more important than including it in the distribution of 2.9.0, which is already code frozen. OK relabeling for 2.10.0 backport?

Yeah SGTM. I've not added `backport 2.9` label, and we can re-label `v2.9.0` -> `v2.10.0`

**Comment by joshuarrrr** [comment]
all tests passed

**Review by joshuarrrr — APPROVED** [review]

**Review by AMoo-Miki — APPROVED** [review]

**Pull request merged**